src/Hazelcast.Net.Linq.Async/Visitors/HzExpressionVisitor.cs

```diff
@@ -32,39 +32,57 @@ public override Expression Visit(Expression? node)
             if (node == null) return node;
 #pragma warning restore CS8603 // Possible null reference return.
 
-            switch ((HzExpressionType)node.NodeType)
+            return (HzExpressionType)node.NodeType switch
             {
-                case HzExpressionType.Map:
-                    return VisitMap((MapExpression)node);
-                case HzExpressionType.Column:
-                    return VisitColumn((ColumnExpression)node);
-                case HzExpressionType.Select:
-                    return VisitSelect((SelectExpression)node);
-                case HzExpressionType.Projection:
-                    return VisitProjection((ProjectionExpression)node);
-                default:
-                    return base.Visit(node);
-            }
+                HzExpressionType.Map => VisitMap((MapExpression)node),
+                HzExpressionType.Column => VisitColumn((ColumnExpression)node),
+                HzExpressionType.Select => VisitSelect((SelectExpression)node),
+                HzExpressionType.Projection => VisitProjection((ProjectionExpression)node),
+                HzExpressionType.Join => VisitJoin((JoinExpression)node),
+                _ => base.Visit(node),
+            };
+        }
+
+        //internal for test purposes
+        internal virtual Expression VisitJoin(JoinExpression node)
+        {
+            var left = Visit(node.Left);
+            var right = Visit(node.Right);
+            var condtition = Visit(node.JoinCondition);
```

Comment:
rename var to `condition`

Reply:
Fixed at https://github.com/hazelcast/hazelcast-csharp-client/pull/732/commits/cea4111d7f8995ee8ea7fad26026d61c33d9d206